### PR TITLE
Add support for IC field

### LINKED ISSUE #6: As a registration admin, I can add a patient's IC into the system, so that I can keep track of their IC for future references.
Sub part of #4

### PR DESCRIPTION
**Pull request by PrestonTYR**
Implement IC class
Update AddCommand, EditCommand and Person to include IC
Closes #6

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -20,6 +21,7 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a person to the address book. "
             + "Parameters: "
+            + PREFIX_IC + "IC "
             + PREFIX_NAME + "NAME "
             + PREFIX_PHONE + "PHONE "
             + PREFIX_EMAIL + "EMAIL "
```
Comment (Wong-ZZ) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:28]: Shouldn't example usage (below) be updated to include the IC field as well?

File: src/main/java/seedu/address/model/person/Ic.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Ic {
```
Comment (Wong-ZZ) [inline src/main/java/seedu/address/model/person/Ic.java:10]: Shouldn't a javadoc comment be included similar to the other models?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 3]
```diff
@@ -48,94 +51,94 @@ public void parse_allFieldsPresent_success() {
         Person expectedPerson = new PersonBuilder(BOB).withTags(VALID_TAG_FRIEND).build();
 
         // whitespace only preamble
-        assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + IC_DESC_BOB + NAME_DESC_BOB + PHONE_DESC_BOB
+            + EMAIL_DESC_BOB + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedPerson));
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:67]: Shouldn't there be a multiple ICs test?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 4]
```diff
@@ -48,94 +51,94 @@ public void parse_allFieldsPresent_success() {
...
         Person expectedPerson = new PersonBuilder(AMY).withTags().build();
-        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY,
-                new AddCommand(expectedPerson));
+        assertParseSuccess(parser, IC_DESC_AMY + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY
+                + ADDRESS_DESC_AMY, new AddCommand(expectedPerson));
     }
 
     @Test
     public void parse_compulsoryFieldMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE);
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:107]: Shouldn't there be a missing IC prefix test?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 5]
```diff
@@ -48,94 +51,94 @@ public void parse_allFieldsPresent_success() {
...
         // all prefixes missing
-        assertParseFailure(parser, VALID_NAME_BOB + VALID_PHONE_BOB + VALID_EMAIL_BOB + VALID_ADDRESS_BOB,
-                expectedMessage);
+        assertParseFailure(parser, VALID_IC_BOB + VALID_NAME_BOB + VALID_PHONE_BOB + VALID_EMAIL_BOB
+                + VALID_ADDRESS_BOB, expectedMessage);
     }
 
     @Test
     public void parse_invalidValue_failure() {
         // invalid name
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:143]: Shouldn't there be an invalid IC test?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 6]
```diff
@@ -76,7 +78,7 @@ public void parse_invalidPreamble_failure() {
         assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
 
         // invalid prefix being parsed as preamble
-        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+        assertParseFailure(parser, "1 l/ string", MESSAGE_INVALID_FORMAT);
     }
 
     @Test
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:94]: Shouldn't there be an invalid IC test?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 7]
```diff
@@ -108,12 +110,12 @@ public void parse_invalidValue_failure() {
...
 
-        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withName(VALID_NAME_AMY)
-                .withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_AMY).withAddress(VALID_ADDRESS_AMY)
-                .withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withIc(VALID_IC_AMY)
+                .withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_AMY)
+                .withAddress(VALID_ADDRESS_AMY).withTags(VALID_TAG_HUSBAND, VALID_TAG_FRIEND).build();
         EditCommand expectedCommand = new EditCommand(targetIndex, descriptor);
 
         assertParseSuccess(parser, userInput, expectedCommand);
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:136]: Shouldn't `parse_oneFieldSpecified_success` and `parse_multipleRepeatedFields_acceptsLast` also have tests for IC?

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 8]
```diff
@@ -41,59 +43,63 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
     @Test
```
Comment (Wong-ZZ) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:47]: Shouldn't there be tests for null and invalid IC?

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 1 above)
Comment (PrestonTYR) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:28]: yep added :D I missed it the first time

**Review by Wong-ZZ — APPROVED** [review]
lgtm

**Pull request merged**